ember-core/src/main/scala/org/http4s/ember/core/Parser.scala

```diff
@@ -325,13 +331,19 @@ private[ember] object Parser {
                   ) ->
                   rest.get
             }
+          } else if (expectNoBody(prelude.status)) {
+            (baseResp -> (Some(finalBuffer): Option[Array[Byte]]).pure[F]).pure[F]
           } else {
-            Body.parseFixedBody(headerP.contentLength.getOrElse(0L), finalBuffer, read).map {
-              case (bodyStream, drain) =>
+            headerP.contentLength
+              .fold(Body.parseUnknownBody(finalBuffer, read))(
```

Comment:
I'm curious should be that behavior be manageable? Though I can't imagine that somebody wants this specific behavior.